lib/Optimizer/Lower.cpp

```diff
@@ -44,17 +44,19 @@ static void replaceAllUsesOfWith(LoweredInfoMap *loweredMap, NodeValue oldNV,
                       oldNV.getNode()->getKind()));
 }
 
-static void lowerAddGradNode(Function *F, LoweredInfoMap *loweredMap,
+static void lowerAddGradNode(Function *F, CompilationContext &cctx,
                              const AddGradNode &node) {
   /// The chain rule for addition:
   /// delta(LHS) = dF/dLHS * delta(OUT) = 1 * delta(OUT)
   /// delta(RHS) = dF/dRHS * delta(OUT) = 1 * delta(OUT)
   auto outG = node.getGradOfOriginalOutputNamedResult();
-  replaceAllUsesOfWith(loweredMap, node.getGradOfInputNamedLHS(), outG);
-  replaceAllUsesOfWith(loweredMap, node.getGradOfInputNamedRHS(), outG);
+  replaceAllUsesOfWith(cctx.loweredInfoMap, node.getGradOfInputNamedLHS(),
```

Comment:
Before landing this I think you should think hard about whether or not you want to log replacement in each place it occurs, or whether you can log one time in `replaceAllUsesOfWith`. If you can do the second then we'll want to pass the context into that method.

Reply:
Great idea. Working on intercepting just that method right now.